### DISCUSSION TITLE
Where can I find Gupta and Suzumura's algorithm?

### DISCUSSION THREAD

**Discussion by ruidazeng** [networkx/networkx#7710] [Q&A] [answered]
Is the 2021 algorithm by Gupta and Suzumura, detailed in their paper "[Finding All Bounded-Length Simple Cycles in a Directed Graph](https://arxiv.org/abs/2105.10094)" implemented in NetworkX? How does `simple_cycles` distinguish between bounded and unbounded length weighted directed cycles?

**Comment by dschult** [comment, marked as answer]
Take a look at [the docs for simple_cycles](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.cycles.simple_cycles.html). It says how to get the Gupta and Suzumura algorithm.

**Reply by ruidazeng** [reply]
Do I have to extract the algorithm from `simple_cycles`?

**Reply by dschult** [reply]
The docs say that "in the bounded case" we use that algorithm. So setting the optional keyword arg `length_bound>1` lets the code use that algorithm. (If you effectively don't want a bound, set it to a really high value.)
